torchvision/csrc/io/decoder/decoder.cpp

```diff
@@ -585,10 +572,10 @@ int Decoder::getFrame(size_t workingTimeInMs) {
 
     result = 0;
 
-    av_packet_unref(&avPacket);
+    av_packet_unref(avPacket);
   }
 
-  av_packet_unref(&avPacket);
+  av_packet_unref(avPacket);
```

Comment:
Two points on this:
1. This second unref was originally here because the loop might terminate via a break. The code is messy and we should refactor it to make it easier to read and have better flow. That's a future NIT.
2. Given that now we call `av_packet_free` call few lines below, should we drop the unref call?

Reply:
1. I agree. IMO the entire API needs a refactor, ideally to be merged with the GPU decoding; This approach  is overly complicated (for historical reasons) for our purposes.
2. Yes if the documentation is to be believed; there are two versions of free packet: `av_packet_free` (which does un-referencing for us), and `av_free_packet` (that doesn't)